include/openssl/lhash.h.in

```diff
@@ -83,14 +87,19 @@ typedef struct lhash_st OPENSSL_LHASH;
 # define LH_LOAD_MULT    256
 
 int OPENSSL_LH_error(OPENSSL_LHASH *lh);
-OPENSSL_LHASH *OPENSSL_LH_new(OPENSSL_LH_HASHFUNC h, OPENSSL_LH_COMPFUNC c);
+OPENSSL_LHASH *OPENSSL_LH_new(OPENSSL_LH_HASHFUNC h, OPENSSL_LH_COMPFUNC c,
+                              OPENSSL_LH_HASHFUNCTHUNK hw,
+                              OPENSSL_LH_COMPFUNCTHUNK cw);
```

Comment:
theres really no way around this, each type of hash table has to have a thunk callback that casts the compare, and hash functions to their respective types to mitigate the ubsan warnings.  This of course wouldn't be a stable branch change, it would have to be master only.

The other option is to strip the typing from the hash tables, which is workable, but completely eliminates the type safety of the hash table (the same applies to STACK_OF, SA_OF, PRIORITY_LIST, etc).

The OTC note in the issue lists out the available options, and why I was hoping to discuss it.

Reply:
No, it can't even be in master yet, not until we decide to release 4.0.

Reply:
yeah, I wasn't speaking clearly above.  This has to wait until master is open to 4.0 PR's

Reply:
It occurs to me that we may be able to work around this.  We can't modify api's, but we could add apis, right?  i.e. if we added  an OPENSSL_lh_set_thunk_funcs function, we could, on create, also call those to set the approrpaite thunking callbacks, correct?  

That would allow us to avoid the ABI breakage and do this sooner

Reply:
I've refactored the LHASH and STACK apis to only add functions for thunking.  It should avoid abi breakage and make this somewhat more palatable for release streams